### DISCUSSION TITLE
Add ability for renovate to force specific branch/rev of given extends

### DISCUSSION THREAD

**Discussion by mdietzer-fn** [renovatebot/renovate#24834] [Suggest an Idea]
### What would you like Renovate to be able to do?

We have a big shared repository of renovate configs (which also acts as the CI pipeline to run renovate).

All our downstream repositories then indicate which configs they want with their local `renovate.json` as follows:
```json
{
    "$schema": "https://docs.renovatebot.com/renovate-schema.json",
    "extends": [
        "ns/renovate",
        "ns/renovate:python",
    ]
}
```

These, of course, make renovate pull those configs from that repo's `main` branch, which is great for the normal use-case.

However, in the case we make changes to the Renovate pipelines themselves, we would like **that** CI (dry-)run to use the just committed hash (to ensure the new configs actually work as we expect them to)

The current way we achieve this goal is by auto-generating `migratePresets` for every single config (and parameterization) we have, such that:

`ns/renovate:python` -> `ns/renovate:python#abcdefg`

`ns/renovate:params(a, b, c)` -> `ns/renovate:params#abcdefg(a, b, c)`

We would like a simpler way to support this use-case.

### If you have any ideas on how this should be implemented, please tell us here.

The simplest solution would be to support wildcards/regex in `migratePresets` which would much simplify this config.

Another good option would be to support "rewrite entire source repo X to use commit Y"

### Is this a feature you are interested in implementing yourself?

Maybe

**Comment by viceice** [comment]
this is already working

`github>abc/foo#1.2.3`

https://docs.renovatebot.com/config-presets/

I'm not sure about the parameters.

@rarkins so mark this as solution?
